pkl-commons-cli/src/main/kotlin/org/pkl/commons/cli/commands/BaseOptions.kt

```diff
@@ -230,7 +237,8 @@ class BaseOptions : OptionGroup() {
       noProject = projectOptions?.noProject ?: false,
       caCertificates = caCertificates,
       httpProxy = proxy,
-      httpNoProxy = noProxy ?: emptyList()
+      httpNoProxy = noProxy ?: emptyList(),
+      colors = if (disableColors) "never" else colors,
```

Comment:
```suggestion
      color = color
```

Reply:
This exists to provide a mechanism for the `test` subcommand to forcefully disable not just colours, but also the injection if ANSI codes completely. I've been having issues because the `pkl test` expected output because they don't travel along the standard output rails. As a consequence the ANSI codes don't get stripped if they're injected.

By providing this escape hatch, I can ensure that colour are disabled _before_ there's any opportunity for the Jansi formatting object to init, because once it's created you can't disable the injection of ANSI codes, only strip them out later. That mixed with the fact that caught errors in `pkl test` are truncated creates a bit of nightmare as the outputs get truncated to slightly different lengths depending on if colours were enabled, regardless of if those ANSI code actually made it to the output.

I tried to figure out where the error truncation etc was happening, so I could modify it to handle ANSI/non-ANSI outputs in an equivalent way, but I haven't been able to track down where it happens.

Reply:
I think if (when?) we switch to a text formatter, we can control this ourselves by passing a no-op text formatter when `--color=never`. Then, we're in full control of how formatting happens rather than mutating global state.

Anyways, I don't know if that's relevant to my suggestion here. `CliBaseOptions` is a data class, and you can always modify it however you want because you get a derived `copy` method.

```kt
val options = baseOptions(myModules).copy(color = Color.NEVER)
```

Reply:
It's only relevant due to how Jansi handles global state. It's caught me out a couple of times because Jansi doesn't handle changes to global state the way you might expect. The final result is what you expect, but the mechanism by which Jansi achieves that result depends on the exact order of operations (mutating global state before or after doing the initial Jansi setup). In most cases it doesn't make any practical difference, but for the specific case of Pkl test it does (which I learned the hard way).

Agree we can just get rid of this hack by implementing a proper formatting API, and having direct control over of control codes are injected.